src/psyclone/tests/psyGen_test.py

```diff
@@ -3559,28 +3556,125 @@ def test_symboltable_can_be_printed():
+    # the argument is accessed
+    sym_v2 = sym_table.lookup("var2")
+    sym_v2.interface = Symbol.Argument(access=Symbol.Access.READWRITE)
+    sym_table.specify_argument_list([sym_v1, sym_v2])
+    assert sym_table.argument_list[1].scope == 'global'
+    assert sym_table.argument_list[1].access == Symbol.Access.READWRITE
+
+
+def test_symboltable_specify_argument_list_errors():
+    ''' Check that supplying specify_argument_list() with Symbols that
+    don't have the correct Interface information raises the expected
+    errors. '''
+    sym_table = SymbolTable()
+    sym_table.add(Symbol("var1", "real", []))
+    sym_table.add(Symbol("var2", "real", []))
+    sym_v1 = sym_table.lookup("var1")
+    # Attempt to say the argument list consists of "var1" which at this
+    # point is just a local variable.
+    with pytest.raises(ValueError) as err:
+        sym_table.specify_argument_list([sym_v1])
+    assert "Symbol 'var1:" in str(err)
+    assert ("is listed as a kernel argument but has no associated "
+            "Interface" in str(err))
+    # Now add an Interface for "var1" but of the wrong type
+    sym_v1.interface = Symbol.FortranGlobal("some_mod")
+    with pytest.raises(ValueError) as err:
+        sym_table.specify_argument_list([sym_v1])
+    assert "Symbol 'var1:" in str(err)
+    assert "has an interface of type '" in str(err)
+
+
+def test_symboltable_argument_list_errors():
+    ''' Tests the internal sanity checks of the SymbolTable.argument_list
+    property. '''
+    sym_table = SymbolTable()
+    sym_table.add(Symbol("var1", "real", []))
+    sym_table.add(Symbol("var2", "real", []))
+    sym_table.add(Symbol("var3", "real",
+                         interface=Symbol.FortranGlobal("some_mod")))
+    # Manually put a local symbol into the internal list of arguments
+    sym_table._argument_list = [sym_table.lookup("var1")]
+    with pytest.raises(ValueError) as err:
+        sym_table._validate_arg_list(sym_table._argument_list)
+    pattern = ("Symbol \'var1.*\' is listed as a kernel argument but has "
+               "no associated Interface")
+    assert re.search(pattern, str(err)) is not None
+    # Check that the argument_list property converts this error into an
+    # InternalError
+    with pytest.raises(InternalError) as err:
+        _ = sym_table.argument_list
+    # Check that we reject a symbol imported from a module
+    with pytest.raises(ValueError) as err:
+        sym_table._validate_arg_list([sym_table.lookup("var3")])
+    # Manually put that symbol into the argument list
+    sym_table._argument_list = [sym_table.lookup("var3")]
+    pattern = (r"Symbol \'var3.*\' is listed as a kernel argument but has an "
+               r"interface of type \'.*\.FortranGlobal\'>")
+    assert re.search(pattern, str(err)) is not None
+    # Check that the argument_list property converts this error into an
+    # InternalError
+    with pytest.raises(InternalError) as err:
+        _ = sym_table.argument_list()
```

Comment:
pylint says `sym_table.argument_list` is not callable, maybe without paranthesis? Also, is the str(err) comparison assert missing?

Reply:
Oops and no, I'd just not bothered :-) I've added it here and above now.